### DISCUSSION TITLE
Roadmap Overview & Development Cycle

### DISCUSSION THREAD

**Discussion by dtrudg** [sylabs/singularity#1629] [Roadmap]
## Introduction

This discussion category will be used to develop the roadmap for future versions of SingularityCE.

Each version will have a specific thread, holding its portion of the roadmap, and where we invite everyone to comment and suggest features for inclusion in the release. The posts holding the roadmap will be edited.

SingularityCE was created in May 2021 as a fork from the (then) hpcng/singularity repository, which has in turn forked to become Apptainer. SingularityCE remains an open source project under the BSD license. [Sylabs](https://sylabs.io/singularity) continues to carry out the majority of Singularity development work and acts as project maintainer, with a strong committment to involving and growing the community of open source users and contributors.

The first SingularityCE release was 3.8.0, on May 26th 2021.

At the time of writing, the current SingularityCE release is 4.1.5, issued on Aug 14th 2024.

When feature requests are fleshed out, and it's agreed which version is a good target, we'll make sure the roadmap matches up with GitHub milestones that help to track the development process.

## Development Cycle

SingularityCE continues to use a 6-month release cycle for minor/major version bumps. The exact timing of releases varies, but we aim to issue a new SingularityCE version in *Jun - Jul* and *Nov - Dec*. Generally, the earlier release will contain a larger range of new features, while the later release focuses on integration / workflow / technical improvements.

**Sylabs disclosure** - Sylabs takes each Nov - Dec release of SingularityCE as the basis of a commercial long-term supported SingularityPRO release. This does have some impact on decisions around the SingularityCE release timelines, but we hope that you will agree that it allows us to sustainably make valuable contributions to the project. Also, any fixes resulting from QA performed as part of this work are incoporated into SingularityCE for the benefit of the open source community.

# Background Information / References

Background information covering the OCI / SIF work in SingularityCE 3.10 -> 4.0 was posted to the SingularityCE Google Group, and at <https://sylabs.io/2022/02/singularityce-4-0-and-beyond>.

**Comment by peci1** [comment]
Could you please comment a bit on the support lifecycle of the released versions? E.g. does the recent 4.0 release mean that no 3.* version will receive support? Or is there some kind of LTS branch?

**Reply by dtrudg** [reply]
LTS is offered by Sylabs via [SingularityPRO](https://sylabs.io/singularity-pro/) which is a commercial offering.

You are correct that the SingularityCE 4.0 release means that SingularityCE 3.11 (and prior versions) no longer receive any updates. Only the latest major.minor SingularityCE release receives bug fixes, security updates etc.
